packages/concerto-tools/test/codegen/fromJsonSchema/cto/jsonSchemaVisitor.js

```diff
@@ -0,0 +1,413 @@
+                }
+            );
+
+        const inferredConcertoModel = Printer.toCTO(
+            inferredConcertoJsonModel.models[0]
+        );
+
+        inferredConcertoModel.should.equal(`namespace com.test@1.0.0
+
+concept arrays {
+  o String[] fruits optional
+  o veggie[] vegetables optional
+}
+
+concept veggie {
+  o String veggieName
+  o Boolean veggieLike
+}`);
+    });
+
+    it('should generate Concerto for for a schema that property modifiers', async () => {
+        const schema = JSON.parse(
+            fs.readFileSync(
+                path.resolve(__dirname, '../cto/data/modifiers-schema.json'),
+                'utf8'
+            )
+        );
+
+        const inferredConcertoJsonModel = JsonSchemaVisitor
+            .parse(schema)
+            .accept(
+                jsonSchemaVisitor,
+                {
+                    ...jsonSchemaVisitorParameters,
+                    pathToDefinitions: ['$defs']
+                }
+            );
+
+        const inferredConcertoModel = Printer.toCTO(
+            inferredConcertoJsonModel.models[0]
+        );
+
+        inferredConcertoModel.should.equal(`namespace com.test@1.0.0
+
+concept geographical_location {
+  o String name default="home" regex=/[\\w\\s]+/ optional
+  o Double latitude
+  o Double longitude range=[-180,180]
+  o Double elevation range=[-11034,] optional
+  o Integer yearDiscovered range=[,2022] optional
+}`);
+    });
+
+    it('should not generate for a simple definition with an unsupported type', async () => {
+        (
+            () => JsonSchemaVisitor
+                .parse({
+                    $schema: 'http://json-schema.org/draft-07/schema#',
+                    type: 'object',
+                    properties: {
+                        foo: {
+                            const: 'value'
+                        }
+                    }
+                })
+                .accept(
+                    jsonSchemaVisitor, { ...jsonSchemaVisitorParameters }
+                )
+        ).should.throw(
+            'Type keyword \'undefined\' in \'foo\' is not supported.'
+        );
+    });
+
+    it(
+        'should not generate when additionalProperties are allowed',
+        async () => {
+            (
+                () => JsonSchemaVisitor
+                    .parse({
+                        $schema: 'http://json-schema.org/draft-07/schema#',
+                        definitions: {
+                            Foo: {
+                                type: 'object',
+                                properties: {},
+                                additionalProperties: true,
+                            }
+                        }
+                    })
+                    .accept(
+                        jsonSchemaVisitor, { ...jsonSchemaVisitorParameters }
+                    )
+            ).should.throw(
+                'Definition \'Foo\' is undefined and unsupported by Concerto.'
+            );
```

Comment:
I think that we can support this scenario now using your `@stringifiedJson` decorator?